### DISCUSSION TITLE
Resizing ZFS Pools

### DISCUSSION THREAD

**Discussion by spflanze** [openzfs/zfs#13745] [General]
I did a highly customized Ubuntu 22.04 installation. The details of how it was done can be downloaded from:
https://www.mediafire.com/file/2w8mdb96tbzslub/Ubuntu_22.04_Root_on_ZFS_Encryption.odt/file

During the installation of Pop!_OS I discovered I had not allocated enough SSD memory for the bpool (boot pool). The RAIDed SSD drives are fully allocated. So I need to increase the bpool at the expense of a little bit of rpool (root pool). Can these allocations be resized? Or will it be necessary to reinstall ZFS, and Ubuntu, with the needed allocation changes?

I started this thread to ask for bpool size recommendations:
https://www.reddit.com/r/pop_os/comments/wh59zx/how_much_should_be_allocate_for_a_pop_s_boot/

**Comment by IvanVolosyuk** [comment]
You can expand zpool but cannot shrink. Unfortunate limitation of ZFS.
